pennylane/tape/qnode.py

```diff
@@ -150,6 +150,7 @@ def __init__(
         self.func = func
         self.device = device
         self.qtape = None
+        self.qfunc_output = None

```

Comment:
Everything here is a direct copy of https://github.com/PennyLaneAI/pennylane/pull/865, so please defer to that for code review.